elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/QueryTranslator.java

```diff
@@ -268,14 +270,29 @@ private String extractJoinClause(Class<?> fromClass,
                         dictionary.getAnnotatedColumnName(
                                 joinClass,
                                 dictionary.getIdFieldName(joinClass)))
-                : extractJoinExpression(join.value(), fromAlias, joinAlias);
+                : getJoinClause(fromClass, fromAlias, joinClass, joinAlias, join.value());
 
         return String.format("LEFT JOIN %s AS %s ON %s",
                 joinSource,
                 joinAlias,
                 joinClause);
     }
 
+    /**
+     * Resolve references to construct a join ON clause.
+     *
+     * @param fromClass parent class
+     * @param fromAlias parent alias
+     * @param joinClass join class
+     * @param joinAlias join alias
+     * @param expr unresolved ON clause
+     * @return string resolved ON clause
+     */
+    private String getJoinClause(Class<?> fromClass, String fromAlias, Class<?> joinClass, String joinAlias,
+                    String expr) {
+        SQLReferenceVisitor visitor = new SQLReferenceVisitor(referenceTable.getMetaDataStore(), fromAlias);
```

Comment:
This works a little different than how other references are resolved.  The SQLReferenceVisitor is normally called at service start and all the references are saved off in the SQLReferenceTable.  From that point onward, only lookups happen in the reference table.  

Is there any reason we can't follow the same approach here?